hack/release-tools/tag-release.sh

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 the Velero contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# This script will do the necessary checks and actions to create a release of Velero.
+# It will first validate that all prerequisites are met, then verify the version string is what the user expects.
+# A git tag will be created and pushed to GitHub, and GoReleaser will be invoked.
+
+# This script is meant to be a combination of documentation and executable.
+# If you have questions at any point, please stop and ask!
+
+# Directory in which the script itself resides, so we can use it for calling programs that are in the same directory.
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+function tag_and_push() {
+    echo "Tagging and pushing $VELERO_VERSION"
+    git tag $VELERO_VERSION
+    git push $VELERO_VERSION
+}
+
+# For now, have the person doing the release pass in the VELERO_VERSION variable as an environment variable.
+# In the future, we might be able to inspect git via `git describe --abbrev=0` to get a hint for it.
+if [[ -z "$VELERO_VERSION" ]]; then
+    printf "The \$VELERO_VERSION environment variable is not set. Please set it with\n\texport VELERO_VERSION=v<version.to.release>\nthen try again."
+    exit 1
+fi
+
+# Make sure the user's provided their github token, so we can give it to goreleaser.
+if [[ -z "$GITHUB_TOKEN" ]]; then
+    printf "The GITHUB_TOKEN environment variable is not set. Please set it with\n\t export GITHUB_TOKEN=<your github token>\n then try again."
+    exit 1
+fi
+
+# Ensure that we have a clean working tree before we let any changes happen, especially important for cutting release branches.
+#if [[ -n $(git status --short) ]]; then 
+#    echo "Your git working directory is dirty! Please clean up untracked files and stash any changes before proceeding."
+#    exit 3
+#fi
```

Comment:
This should be uncommented to avoid publishing release built from local changes.